src/sentry/shared_integrations/response/base.py

```diff
@@ -10,28 +14,35 @@
 class BaseApiResponse:
     text = ""
 
-    def __init__(self, headers=None, status_code=None):
+    def __init__(
+        self,
+        headers: Mapping[str, str] | None = None,
+        status_code: int | None = None,
+    ) -> None:
         self.headers = headers
         self.status_code = status_code
 
-    def __repr__(self):
-        return "<{}: code={}, content_type={}>".format(
-            type(self).__name__,
-            self.status_code,
-            self.headers.get("Content-Type", "") if self.headers else "",
-        )
+    def __repr__(self) -> str:
+        content_type = (self.headers or {}).get("Content-Type", "")
+        return f"<{type(self).__name__}: code={self.status_code}, content_type={content_type}>"
```

Comment:
personal opinion but the old code was more readable to me. perhaps move all formatting arguments in their own variables